src/platform/packages/shared/kbn-cell-actions/.eslintrc.js

```diff
@@ -7,54 +7,127 @@
+ * - First element: severity (required)
+ * - Subsequent elements: restrictions in either:
+ *   a) Legacy format (strings/objects)
+ *   b) Modern format (configuration object with paths/patterns)
+ */
 
-const TEST_FILE_PATTERNS = [
-  '*.mock.{js,ts,tsx}',
-  '*.test.{js,ts,tsx}',
-  '*.test.helpers.{js,ts,tsx}',
-  '*.stub.{js,ts,tsx}',
-  'mock.{js,ts,tsx}',
-  '_stubs.{js,ts,tsx}',
-  '{testHelpers,test_helper,test_utils}.{js,ts,tsx}',
+/** @type {import('eslint').Linter.Config} */
+const rootConfig = require('../../../../../.eslintrc'); // eslint-disable-line @kbn/imports/no_boundary_crossing
+
+/** @type {Array.<RestrictedImportPath>} */
+const RESTRICTED_IMPORTS = [
+  {
+    name: 'enzyme',
+    message: 'Please use @testing-library/react instead',
+  },
 ];
 
-module.exports = {
-  root: false,
-
-  extends: ['../../../../../.eslintrc.js'],
-
-  overrides: [
-    {
-      files: [
-        ...TEST_DIRECTORIES.map((dir) => `**/${dir}/**/*.{js,ts,tsx}`),
-        ...TEST_FILE_PATTERNS.map((pattern) => `**/${pattern}`),
-      ],
-      rules: {
-        'no-restricted-imports': [
-          'error',
-          {
-            name: 'enzyme',
-            message: 'Please use @testing-library/react instead',
-          },
-        ],
+const clonedRootConfig = structuredClone(rootConfig);
+
+const overridesWithNoRestrictedImportRule = clonedRootConfig.overrides.filter(
+  (override) => override.rules && 'no-restricted-imports' in override.rules
+);
+
+overridesWithNoRestrictedImportRule.forEach((override) => {
+  /** @type {NoRestrictedImportsRuleConfig} */
+  const rule = override.rules['no-restricted-imports'];
+
+  if (Array.isArray(rule) && rule.length >= 2) {
+    const [severity, ...rawOptions] = rule;
+
+    /** @type {RestrictedImportOptions} */
+    const modernConfig = { paths: [], patterns: [] };
+
+    // Normalize all inputs into modern config format
+    rawOptions.forEach((opt) => {
+      if (typeof opt === 'string' || 'name' in opt) {
+        modernConfig.paths.push(opt);
+      } else if ('paths' in opt || 'patterns' in opt) {
+        modernConfig.paths.push(...(opt.paths || []));
+        modernConfig.patterns.push(...(opt.patterns || []));
+      }
+    });
+
+    // Dynamic duplicate removal for all restricted imports
+    const existingPaths = modernConfig.paths.filter(
+      (existing) =>
+        !RESTRICTED_IMPORTS.some((restriction) =>
+          typeof existing === 'string'
+            ? existing === restriction.name
+            : existing.name === restriction.name
+        )
+    );
+
+    /** @type {NoRestrictedImportsRuleConfig} */
+    const newRuleConfig = [
+      severity,
+      {
+        paths: [...existingPaths, ...RESTRICTED_IMPORTS],
+        patterns: modernConfig.patterns,
       },
-    },
-  ],
+    ];
+
+    override.rules['no-restricted-imports'] = newRuleConfig;
+  }
+});
+
+module.exports = {
+  overrides: overridesWithNoRestrictedImportRule,
```

Comment:
This is the second attempt. It almost worked. The problem with this was
very subtle but still pretty impactful.

Since I am redeclaring matching override blocks from root .eslintrc.js into nested .eslintrc.js I also move in the overrides[].files property

But the problem is those paths are manifesting the path from root config and they don't work with this nested config. Thus entire chunks of overrides are disabled because they never match.

Thus we come to attempt #3 with the next commit